core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java

```diff
@@ -552,41 +563,51 @@ private boolean drainQueue() {
                         eventProcessed = true;
                     }
                 }
+
+                // by not releasing cursor we force the sequence to return us the same value over and over
+                // until cursor value is released
                 if (eventProcessed) {
                     sequence.done(cursor);
                 }
             }
         }
 
-        private void handleEventException(LineTcpMeasurementEvent event, Parser parser, CairoException ex) {
-            LOG.error()
-                    .$("could not create parser, measurement will be skipped [jobName=").$(jobName)
-                    .$(", table=").$(event.getTableName())
-                    .$(", ex=").$(ex.getFlyweightMessage())
-                    .$(", errno=").$(ex.getErrno())
-                    .$(']').$();
-            Misc.free(parser);
-            parserCache.remove(event.getTableName());
-        }
-
         private void processNextEvent(LineTcpMeasurementEvent event) {
-            Parser parser = parserCache.get(event.getTableName());
+            final int index = parserCache.keyIndex(event.getTableName());
+            Parser parser = null;
             try {
-                if (null != parser) {
+                if (index < 0) {
+                    parser = parserCache.valueAt(index);
                     parser.processEvent(event);
                 } else {
                     parser = new Parser();
                     parser.processFirstEvent(engine, securityContext, event);
                     LOG.info().$("created parser [jobName=").$(jobName).$(" table=").$(event.getTableName()).$(']').$();
-                    parserCache.put(Chars.toString(event.getTableName()), parser);
+                    parserCache.putAt(index, Chars.toString(event.getTableName()), parser);
                 }
             } catch (CairoException ex) {
-                handleEventException(event, parser, ex);
+                LOG.error()
+                        .$("could not create parser, measurement will be skipped [jobName=").$(jobName)
+                        .$(", table=").$(event.getTableName())
+                        .$(", ex=").$(ex.getFlyweightMessage())
+                        .$(", errno=").$(ex.getErrno())
+                        .$(']').$();
+
+                Misc.free(parser);
+
+                if (index < 0) {
+                    parserCache.removeAt(index);
+                }
             }
         }
 
         private boolean processRebalance(LineTcpMeasurementEvent event) {
             if (event.rebalanceToThreadId == id) {
+                // This thread is now a declared owner of the table, but it can only become actual
+                // owner when "old" over is fully done. This is a volatile variable on the event, used by both threads
```

Comment:
// owner when "old" owner is fully done. This is a volatile variable on the event, used by both threads